spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -71,36 +79,38 @@ public static void main(String[] args) throws Exception {
 			System.err.println("Either the 'modules' system property or 'MODULES' environment variable is required.");
 			System.exit(1);
 		}
-		String moduleHome = System.getProperty("module.home");
-		if (moduleHome == null) {
-			moduleHome = System.getenv("MODULE_HOME");
-		}
-		if (moduleHome == null) {
-			moduleHome = DEFAULT_MODULE_HOME;
-		}
-		ModuleLauncher launcher = new ModuleLauncher(new File(moduleHome));
+		ModuleLauncher launcher = new ModuleLauncher();
 		launcher.launch(modules.split(","), args);
 	}
 
 
 	private static class ModuleLaunchTask implements Runnable {
 
-		private final File directory;
+		private final ModuleResolver moduleResolver;
+
+		private final String groupId;
+
+		private final String artifactId;
 
-		private final String module;
+		private final String version;
 
 		private final String[] args;
 
-		ModuleLaunchTask(File directory, String module, String[] args) {
-			this.directory = directory;
-			this.module = module;
+		ModuleLaunchTask(ModuleResolver moduleResolver, String module, String[] args) {
+			this.moduleResolver = moduleResolver;
+			String[] tokens = module.split(":");
+			Assert.isTrue(tokens.length == 3, "required module format is: 'groupId:artifactId:version'");
+			this.groupId = tokens[0];
+			this.artifactId = tokens[1];
+			this.version = tokens[2];
 			this.args = args;
 		}
 
 		@Override
 		public void run() {
 			try {
-				JarFileArchive jarFileArchive = new JarFileArchive(new File(directory, module));
+				Resource resource = moduleResolver.resolve(groupId, artifactId, version, "exec", "jar");
```

Comment:
The args are in a different order in https://github.com/spring-cloud/spring-cloud-stream/pull/47
 moduleResolver.resolve(groupId, artifactId,  "jar", "exec", version);